canonical-websites/snapcraft.io
# Account details and snaps list

Implements https://github.com/canonicalltd/snapcraft-design/issues/302
and https://github.com/canonicalltd/snapcraft-design/issues/292

# QA
- Pull the branch
- `./run`
- Visit http://0.0.0.0:8004/account it should redirect to /account/snaps
- Visit http://0.0.0.0:8004/account/snaps you should see your list of snaps
- Visit http://0.0.0.0:8004/account/details you should see your account information

To QA empty snaps page:
- edit `modules/publisher/views.py:101` and add:
```
snaps_list = {}
registered_snaps = {}
```

To QA a single snap:
- edit `snaps_list = {}` added above ^
```
snaps_list = {'name-of-snap': snaps_list['name-of-snap']}
```

Demo links:
http://snapcraft.io-pr-421.run.demo.haus/account
http://snapcraft.io-pr-421.run.demo.haus/account/details
http://snapcraft.io-pr-421.run.demo.haus/account/snaps

## Missing

### /account/snaps No snaps

- [x] Snapcraft bird (though I imagine it will be the same outcome as with the 404 page - the bird shouldn't be used like that) @lyubomir-popov please check @Greggless wireframes: https://github.com/canonicalltd/snapcraft-design/issues/302

### /account/snaps First snap

- [ ] Data for 'You've just released your first snap to the testing "X" channel.' where X is the channel.
- [ ] Ability to toggle privacy of snap (conversation with @cprov needed/ started)
- [x] Up to date image of 'Market' page as will be on launch

### /account/snaps n snaps

- [x] Owner field - should be live in the API by 22-03-2018 (tomorrow at the time of writing)

## Screenshots

### /account/details
![screenshot-2018-4-19 account details linux software in the snap store](https://user-images.githubusercontent.com/479384/38982662-5c9b467a-43ba-11e8-90b8-b2e8507ffcb9.png)

### /account/snaps No snaps
![screenshot-2018-4-19 my snaps linux software in the snap store](https://user-images.githubusercontent.com/479384/38982670-6138ccfc-43ba-11e8-8483-fa79bacdd85c.png)

### /account/snaps First snap
![screenshot-2018-4-19 my snaps linux software in the snap store 4](https://user-images.githubusercontent.com/479384/38985679-ff436428-43c1-11e8-956f-d2ce92d094a9.png)

### /account/snaps n snaps
![screenshot-2018-4-19 my snaps linux software in the snap store 2](https://user-images.githubusercontent.com/479384/38982688-6c229ba2-43ba-11e8-9ffd-0156a050e618.png)

### /account/snaps Registered names
![screenshot-2018-4-19 my snaps linux software in the snap store 3](https://user-images.githubusercontent.com/479384/38982694-702ad250-43ba-11e8-8d26-b364283224e9.png)
